components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/util/I18nEmailUtil.java

```diff
@@ -281,4 +285,44 @@ public static String getNotificationLocale() {
                 ? IdentityUtil.getProperty(I18nMgtConstants.NOTIFICATION_DEFAULT_LOCALE)
                 : I18nMgtConstants.DEFAULT_NOTIFICATION_LOCALE;
     }
+
+    /**
+     * Get the notification template subject, body & footer contents as a byte array.
+     *
+     * @param notificationTemplate  Notification template to get the content
+     * @return                      Byte array of the content
+     */
+    public static byte[] getContentByteArray(NotificationTemplate notificationTemplate) {
+
+        String[] templateContent = new String[]{notificationTemplate.getSubject(), notificationTemplate.getBody(),
+                notificationTemplate.getFooter()};
+        String content = new Gson().toJson(templateContent);
+        return content.getBytes(StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Read the content stream and set the subject, body & footer of the notification template.
+     *
+     * @param contentStream                 InputStream of the content
+     * @param notificationTemplateResult    Notification template to set the content
+     * @throws SQLException
+     */
+    public static void setContent(InputStream contentStream, NotificationTemplate notificationTemplateResult) throws
+            SQLException {
+
+        try {
+            byte[] contentByteArray = contentStream.readAllBytes();
+            String content = new String(contentByteArray, StandardCharsets.UTF_8);
+            String[] templateContent = new Gson().fromJson(content, String[].class);
+            if (templateContent != null && templateContent.length == 3) {
+                notificationTemplateResult.setSubject(templateContent[0]);
+                notificationTemplateResult.setBody(templateContent[1]);
+                notificationTemplateResult.setFooter(templateContent[2]);
+            } else {
+                throw new SQLException("Invalid content data.");
```

Comment:
Is it ok to throw a SQL Exception here?

Reply:
Most simple answer is "Yes".. 😃
Let me know if you have any specific concerns..

Reply:
To add more context, this util is used in the DAO impls.. Specifically this is used as a instruction of the `RowMapper.mapRow()` functional param of the jdbc templates. The `RowMapper.mapRow()` only allowed to throw exceptions in the type of SQLException.